plugins/catalog-backend-module-azure-blobstorage/src/provider/AzureBlobStorageEntityProvider.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  EntityProvider,
+  EntityProviderConnection,
+  locationSpecToLocationEntity,
+} from '@backstage/plugin-catalog-node';
+import { Logger } from 'winston';
+import { Config } from '@backstage/config';
+import { PluginTaskScheduler, TaskRunner } from '@backstage/backend-tasks';
+import { readAzureBlobStorageConfigs } from './config';
+import {
+  AzureBlobStorageIntegration,
+  ScmIntegrations,
+} from '@backstage/integration';
+import { LocationSpec } from '@backstage/plugin-catalog-common';
+import { BlobServiceClient, ContainerClient } from '@azure/storage-blob';
+import * as uuid from 'uuid';
+import { AzureBlobStorageConfig } from './types';
+
+/**
+ * Provider which discovers catalog files (any name) within Azure BlobStorage
+ *
+ * Use `AzureStorageEntityProvider.fromConfig(...)` to create instances.
+ *
+ * @public
+ */
+export class AzureBlobStorageEntityProvider implements EntityProvider {
+  private readonly storageClient: BlobServiceClient;
+  private readonly logger: Logger;
+  private readonly scheduleFn: () => Promise<void>;
+  private readonly integration: AzureBlobStorageIntegration;
+  private connection?: EntityProviderConnection;
+
+  static fromConfig(
+    configRoot: Config,
+    options: {
+      logger: Logger;
+      schedule?: TaskRunner;
+      scheduler?: PluginTaskScheduler;
+    },
+  ): AzureBlobStorageEntityProvider[] {
+    const providerConfigs = readAzureBlobStorageConfigs(configRoot);
+    if (!options.schedule && !options.scheduler) {
+      throw new Error('Either schedule or scheduler must be provided.');
+    }
+
+    return providerConfigs.map(providerConfig => {
+      const integration = ScmIntegrations.fromConfig(
+        configRoot,
+      ).azureBlobStorage.byHost(
+        `${providerConfig.accountName}.blob.core.windows.net`,
+      );
+
+      if (!integration) {
+        throw new Error(
+          `No integration found for azureBlobStorage-provider:${providerConfig.accountName}.`,
```

Comment:
hm here the account name is after the colon, not the id or the grouping key - same below - this seems confusing